src/conduit/api.cljs

```diff
@@ -22,17 +23,34 @@
   (str "https://conduit.productionready.io/api/" path))
 
 (defn- parse-body [res]
-  (-> (:body res)
+  (-> res
       js/JSON.parse
       (js->clj :keywordize-keys true)))
 
+(defn- ->xhr [uri xhr-fn params]
+  (-> uri
+      (xhr-fn params)
+      (p/then (fn [{status :status body :body :as response}]
+                (condp = status
+                  status/ok (p/resolved (parse-body body))
+                  (p/rejected (parse-body body)))))))
+
 (defn fetch
```

Comment:
Since now we have so much parameters it would be better to pass a single map instead with all options.